File: backend/src/main/java/com/kiwi/features/quests/data/QuestDTO.java
package com.kiwi.features.quests.data;

import lombok.*;
import java.util.List;

@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder @EqualsAndHashCode @ToString
public class QuestDTO {
    private int questId;
    private String name;
    private String description;
    private int experience;
    private int icon;
    private String status;
    private List<SubquestDTO> subquests;
}

File: mobile/app/src/main/java/com/bellako/kiwi/features/quests/data/QuestDTO.kt
package com.bellako.kiwi.features.quests.data

data class QuestDTO(
    val questId: Int,
    val name: String,
    val description: String,
    val experience: Int,
    val icon: Int,
    val status: String,
    val subquests: List<SubquestDTO>,
    val onCompletedEvent: String,
    val onCompletedEntityId: Int,
)
